data/data-pipeline/data_pipeline/etl/sources/tribal_overlap/etl.py

```diff
@@ -0,0 +1,199 @@
+    NAME = "tribal_overlap"
+    GEO_LEVEL = ValidGeoLevel.CENSUS_TRACT
+
+    PUERTO_RICO_EXPECTED_IN_DATA = False
+    ALASKA_AND_HAWAII_EXPECTED_IN_DATA = True
+    EXPECTED_MISSING_STATES = [
+        "10",
+        "11",
+        "13",
+        "15",
+        "17",
+        "18",
+        "21",
+        "24",
+        "33",
+        "34",
+        "39",
+        "50",
+        "51",
+        "54",
+    ]
+
+    # A Tribal area that requires some special processing.
+    ANNETTE_ISLAND_TRIBAL_NAME = "Annette Island LAR"
+
+    # Define these for easy code completion
+    def __init__(self):
+        self.COLUMNS_TO_KEEP = [
+            self.GEOID_TRACT_FIELD_NAME,
+            field_names.COUNT_OF_TRIBAL_AREAS_IN_TRACT,
+            field_names.PERCENT_OF_TRIBAL_AREA_IN_TRACT,
+            field_names.NAMES_OF_TRIBAL_AREAS_IN_TRACT,
+        ]
+
+        self.output_df: pd.DataFrame
+        self.census_tract_gdf: gpd.GeoDataFrame
+        self.tribal_gdf: gpd.GeoDataFrame
+
+    @staticmethod
+    def _create_string_from_list(series: pd.Series) -> str:
+        """Helper method that creates a sorted string list (for tribal names)."""
+        str_list = series.tolist()
+        str_list = sorted(str_list)
+        return ", ".join(str_list)
+
+    def extract(self) -> None:
+        self.census_tract_gdf = get_tract_geojson()
+        self.tribal_gdf = get_tribal_geojson()
+
+    def transform(self) -> None:
+        logger.info("Starting tribal overlap transforms.")
+
+        # First, calculate whether tracts include any areas from the Tribal areas,
+        # for both the points in AK and the polygons in the continental US (CONUS).
+        tribal_overlap_with_tracts = add_tracts_for_geometries(
+            df=self.tribal_gdf, tract_data=self.census_tract_gdf
+        )
+
+        tribal_overlap_with_tracts = tribal_overlap_with_tracts.groupby(
+            [self.GEOID_TRACT_FIELD_NAME]
+        ).agg(
+            {
+                field_names.TRIBAL_ID: "count",
+                field_names.TRIBAL_LAND_AREA_NAME: self._create_string_from_list,
+            }
+        )
+
+        tribal_overlap_with_tracts = tribal_overlap_with_tracts.reset_index()
+
+        tribal_overlap_with_tracts = tribal_overlap_with_tracts.rename(
+            columns={
+                field_names.TRIBAL_ID: field_names.COUNT_OF_TRIBAL_AREAS_IN_TRACT,
+                field_names.TRIBAL_LAND_AREA_NAME: field_names.NAMES_OF_TRIBAL_AREAS_IN_TRACT,
+            }
+        )
+
+        # Second, calculate percentage overlap.
+        # Drop the points from the Tribal data (because these cannot be joined to a
+        # (Multi)Polygon tract data frame)
+        tribal_gdf_without_points = self.tribal_gdf[
+            self.tribal_gdf.geom_type != "Point"
+        ]
+
+        # Create a measure for the entire census tract area
+        with warnings.catch_warnings():
+            # Note: geopandas will (helpfully) warn that area is not a useful metric
```

Comment:
I ran the line and got this warning:

`UserWarning: `keep_geom_type=True` in overlay resulted in 348 dropped geometries of different geometry types than df1 has. Set `keep_geom_type=False` to retain all geometries`

That was not the warning I expected (I thought it would complain about projected vs geographic CRS. Both DFs have CRSes (they're both in WSG 84/EPSG:4326), so there wouldn't be a mismatch. Any idea what that warning is about? Also I feel like the comment suggests something else is going on.

Reply:
Oh shoot! That is an error I missed. I built this using a subset file of the tracts to improve speed of development, it must not have been present in the subset I used. 

Overlay requires the same geometry types, but it treats MultiPolygon and Polygon as compatible. 

Because we drop the points in line 101 to create `tribal_gdf_without_points`, I wasn't expecting warnings. It turns out that there are some Tribal areas with a "none" geography type. I'm updating them now to fix! 

I'm also more aggressively restricting the filter so it doesn't suppress other helpful warnings like this in the future.

Nice catch @mattbowen-usds !!

Reply:
@mattbowen-usds -- I honestly have no idea what's going on here. I pushed some new code that removes both the Points and the `None` from tribal_gdf. 

But, the warning is still there, even though it should be polygon/multipolygon to polygon/multipolygon, which should be fine.  

Further, it says `348 dropped geometries` -- which is more than the rows in `tribal_gdf_without_points`! 

If this is urgent to get in Monday, do you mind investigating a little bit while I'm out?

Reply:
From my commit

> I looked into this a bit, and in general the geometry type mismatch
> changes very little about the calculation; we have a mix of
> multipolygons and polygons. The fastest thing to do is just not keep
> geom type; I did some runs with it set to both True and False, and
> they're the same within 9 digits of precision. Logically we just want to
> overlaps, regardless of how the actual geometries are encoded between
> the frames, so we can in this case ignore the geom types and feel OKAY.


I changed it to ignore the geometry types because that is what makes sense to me after really digging into it, and also it basically doesn't matter; whatever was dropping before is a TINY (< .00000001%) amount of actual area.